라우팅을 활용하여 레스토랑 상세 페이지 구현

SeaFalcon opened the pull request:
이번주 너무너무 게을렀습니다. 늦어서 죄송해요.

아직 완성은 못했고 작업 중입니다.

TDD 방식에 맞춰서 하고있는지만 봐주시면 감사하겠습니다 !

SeaFalcon committed:
App에서 컴포넌트 RestaurantsPage로 분리

SeaFalcon committed:
[RED+GREEN] App.jsx - RestaurantsPage 렌더 테스트 추가

- react-router-dom 설치
- webpack historyApiFallback - index.html 설정

SeaFalcon committed:
[Refactoring] App.jsx - RestaurantsPage 렌더 테스트 리펙토링

SeaFalcon committed:
[RED] App.jsx - HomePage 렌더 테스트 추가

SeaFalcon committed:
[Green] App.jsx - HomePage 렌더 테스트 구현

SeaFalcon committed:
[Refactoring] App.jsx - HomePage 렌더 테스트 리펙토링

- HomePage 파일 분리, html 레이아웃 작성
- HomePage red + green 테스트 작성

SeaFalcon committed:
[RED] App.jsx - AboutPage 렌더 테스트 추가

SeaFalcon committed:
[Green] App.jsx - AboutPage 렌더 테스트 구현

SeaFalcon committed:
[Refactoring] App.jsx - AboutPage 렌더 테스트 리펙토링

    - AboutPage 파일 분리, html 레이아웃 작성
    - AboutPage red + green 테스트 작성

SeaFalcon committed:
[Red] App.jsx - NotFoundPage 렌더 테스트 추가

SeaFalcon committed:
[Green] App.jsx - NotFoundPage 렌더 테스트 구현

SeaFalcon committed:
[Refactoring] App.jsx - NotFoundPage 렌더 테스트 리펙토링

- NotFoundPage 파일 분리, html 레이아웃 작성
- NotFoundPage red + green 테스트 작성

SeaFalcon committed:
App.jsx 헤더 추가

SeaFalcon committed:
[Red] RestaurantsContainer 항목 클릭 테스트 추가

SeaFalcon committed:
[Red] actions.js loadRestaurantDetail 테스트 추가

- RestaurantsContainer의 항목을 클릭하면 실행되는 함수 추가를 위함

SeaFalcon committed:
[Red] api.js fetchRestaurantDetail 테스트 추가

- restaurant mock 데이터 추가
- action 함수 추가

SeaFalcon committed:
[Green] api.js fetchRestaurantDetail 테스트 구현

SeaFalcon committed:
[Refactoring] api.js fetchRestaurantDetail 테스트 리펙토링

SeaFalcon committed:
[Green] actions.js loadRestaurantDetail 테스트 구현

SeaFalcon committed:
[Green] RestaurantsContainer 항목 클릭 테스트 구현

SeaFalcon committed:
[Red] reducer setRestaurant 테스트 추가

SeaFalcon committed:
[Green] reducer setRestaurant 테스트 구현

SeaFalcon committed:
[Refactoring] reducer setRestaurant 테스트 리펙토링

SeaFalcon committed:
[Red] App.jsx - RestaurantDetailPage 렌더 테스트 추가

SeaFalcon committed:
[Green] App.jsx - RestaurantDetailPage 렌더 테스트 구현

SeaFalcon committed:
[Refactiring] App.jsx - RestaurantDetailPage 렌더 테스트 리팩토링

- index.html src 수정
- App.jsx exact 추가
- RestaurantsDetailContainer 페이지 리펙토링

wholemann reviewed (commented):
잘 하고 계시네요.

우리가 TDD를 왜 하는지 잊으면 안 됩니다.

우리가 모르는 지식을 빠르게 발견하기 위해 TDD를 하는 것이지

TDD로 코딩하기 위해 TDD를 하는 건 아닙니다.

TDD는 문제를 발견하고 올바르게 해결하기 위한 방법론으로 생각해야 합니다.

TDD를 잘하고 있는지 의문이 자꾸 든다면

내가 TDD를 통해 무엇을 해결하고 싶었는지를 돌이켜 봐야 합니다.

몇 가지 아쉬운 부분은 comment 남겨드릴게요:)

wholemann commented on src/RestaurantDetailPage.test.jsx:
```diff
@@ -0,0 +1,57 @@
+
+import RestaurantDetailPage from './RestaurantDetailPage';
+
+import YANGCHEON_JUGA from '../fixtures/restaurant';
```
이건 상수가 아니에요. 그냥 객체입니다. 대문자로 표기하는 건 옳지 않죠.

SeaFalcon committed:
전체적인 리팩토링

- RestaurantsDetailPage를 Container와 Presentation으로 분리
- 상수 제거 (상수를 제대로 이해하고 쓰기)
- 테스트 리팩토링 (의도 더 드러내기)

SeaFalcon commented:
전체적으로 리팩토링 진행해보았습니다.

wholemann commented on src/RestaurantDetailContainer.jsx:
```diff
@@ -0,0 +1,18 @@
+
+import { useSelector } from 'react-redux';
+
+import { get } from './utils';
+
+import RestaurantDetail from './RestaurantDetail';
+import LoadingPage from './LoadingPage';
```
```suggestion
import RestaurantDetail from './RestaurantDetail';
import LoadingPage from './LoadingPage';

import { get } from './utils';
```

순서에 주의해주세요.

wholemann reviewed (approved):
살짝 아쉬운 부분만 comment 하나 남겨드렸습니다.

Merge해도 무리가 없겠네요. 잘하셨어요:)

나머진 아샬님 피드백 강의를 통해 채우실 수 있을 겁니다.

무엇보다 회고를 통해 돌아보는 시간을 갖는 게 제일 중요하죠.
